### PR TITLE
FIM System tests: 0202 - Realtime monitoring

### PR DESCRIPTION
**Pull request by grg121**
Closes https://github.com/wazuh/wazuh-qa/issues/531

Ready for review. I'm getting a lot of error generating alerts with realtime when the scan frequency is high. I'm investigating it but meanwhile, you could review the changes.

Best regards.

### CODE REVIEW

File: tests/system/fim/common_tasks/verify_alerts_elasticsearch.py [hunk 1]
```diff
@@ -76,8 +76,8 @@ def makeQuery(query, Elastic, index_name):
         help="Sleep time between retries", default="60"
     )
     parser.add_argument(
-        "-w", "--whodata", type=bool, required=False, dest='whodata_query',
-        help="Enable whodata queries", default="False"
```
Comment (grg121) [inline tests/system/fim/common_tasks/verify_alerts_elasticsearch.py]: This is not correct, "False" is a string and is interpreted by Python3 as `True` when casting to a boolean.

File: tests/system/fim/common_tasks/verify_alerts_json.py [hunk 2]
```diff
@@ -105,16 +105,16 @@ def main():
                             help="Output path for missing files alerts.",
                             default="debug_missing_file_alerts.log")
 
-        parser.add_argument("-t", "--timeout", type=int, required=False, dest='time_gap',
```
Comment (grg121) [inline tests/system/fim/common_tasks/verify_alerts_json.py]: to avoid misunderstanding

File: tests/system/fim/common_tasks/verify_alerts.yml [hunk 3]
```diff
@@ -3,28 +3,28 @@
...
-    when: 
+    when:
       - inventory_hostname in groups['masters']
 
   - name: "Verify alert json log | {{ event }}"
     script: >
-      verify_alerts_json.py -i {{ item.path }} -e {{ event }} 
-      -o {{ missing_alerts_json_path }}
+      verify_alerts_json.py -i {{ item.path }} -e {{ event }}
+      -o {{ missing_alerts_json_path }} {{ alert_verification_extra_arguments }}
```
Comment (jm404) [inline tests/system/fim/common_tasks/verify_alerts.yml]: ``{{ elastic_verification_extra_arguments }}`` is also an alert verification, please consider changing it to `{{ json_verification_extra_arguments }} as it's related to `alerts.json` verification

File: tests/system/fim/common_tasks/verify_alerts_elasticsearch.py (hunk 1 above)
Comment (jm404) [reply tests/system/fim/common_tasks/verify_alerts_elasticsearch.py]: Why remove the `type` parameter?

Comment (jm404) [reply tests/system/fim/common_tasks/verify_alerts_elasticsearch.py]: I checked `default=False` without quotes and returned False, also when declaring `-w True` returned True.

Please consider leaving `type` parameter and setting the default to `False`

File: tests/system/fim/common_tasks/verify_alerts_json.py (hunk 2 above)
Comment (jm404) [reply tests/system/fim/common_tasks/verify_alerts_json.py]: Parameters are standardized across scripts, so I think we should keep the `-t` parameter, at least for now. Anyway the description can be improved, maybe `--time_sleep` would be better.

File: tests/system/fim/common_tasks/verify_alerts_json.py [hunk 4]
```diff
@@ -105,16 +105,16 @@ def main():
...
-                            help="time gap between alerts.json alerts acquisitions. default: 60s",
-                            default="60")
+        parser.add_argument("-s", "--sleep", type=int, required=False, dest='sleep_time',
+          help="Sleep time between retries", default="60"
+        )
 
         parser.add_argument("-r", "--retry", type=int, required=False, dest='retry_count',
                             help="reading attempts on stopped alerts. default: 4 attemps",
                             default="4")
-        parser.add_argument("-w", "--whodata", type=bool, required=False, dest='whodata_check',
```
Comment (jm404) [inline tests/system/fim/common_tasks/verify_alerts_json.py]: Related to https://github.com/wazuh/wazuh-qa/pull/549/files#r384851561

File: tests/system/fim/common_tasks/verify_alerts.yml (hunk 3 above)
Comment (grg121) [reply tests/system/fim/common_tasks/verify_alerts.yml]: But what if those scripts accept different parameters in the future?

File: tests/system/fim/common_tasks/verify_alerts_elasticsearch.py (hunk 1 above)
Comment (grg121) [reply tests/system/fim/common_tasks/verify_alerts_elasticsearch.py]: Yes, default=False should do the trick, but the correct what to store booleans using arparse and flags is using `store_true` or `store_false`. These are special cases of 'store_const' used for storing the values True and False respectively. In addition, they create default values of False and True respectively.

File: tests/system/fim/common_tasks/verify_alerts_json.py (hunk 2 above)
Comment (grg121) [reply tests/system/fim/common_tasks/verify_alerts_json.py]: Oh, right! But that parameter is `-s` in `verify_alerts_elasticsearch`

File: tests/system/fim/common_tasks/verify_alerts_json.py (hunk 4 above)
Comment (grg121) [reply tests/system/fim/common_tasks/verify_alerts_json.py]: Same answer

File: tests/system/fim/common_tasks/verify_alerts_elasticsearch.py (hunk 1 above)
Comment (jm404) [reply tests/system/fim/common_tasks/verify_alerts_elasticsearch.py]: ` but the correct what to store booleans using arparse and flags is using store_true or store_false` Please link some references so we can have as an argument to propagate this change to others scenarios. Thanks! :+1:

File: tests/system/fim/common_tasks/verify_alerts_json.py (hunk 2 above)
Comment (jm404) [reply tests/system/fim/common_tasks/verify_alerts_json.py]: You are right, I believed we had it as `-t` in `verify_alerts_elasticsearch`. Let's keep the `-s` option.

File: tests/system/fim/common_tasks/verify_alerts.yml (hunk 3 above)
Comment (jm404) [reply tests/system/fim/common_tasks/verify_alerts.yml]: The `{{ elastic_verification_extra_arguments }}` will accept multiple parameters and will be declared using `set_fact`. E.G:

```yml
set_fact:
   elastic_verification_extra_arguments: "-s 60 -w True"
```

File: tests/system/fim/common_tasks/verify_alerts_json.py (hunk 4 above)
Comment (jm404) [reply tests/system/fim/common_tasks/verify_alerts_json.py]: Same response, check it please

File: tests/system/fim/common_tasks/verify_alerts_elasticsearch.py (hunk 1 above)
Comment (grg121) [reply tests/system/fim/common_tasks/verify_alerts_elasticsearch.py]: Sure, check https://docs.python.org/3/library/argparse.html
As I said, `default=False` with `type` will work but that library has a special way to add booleans (search for `store_true` for more info) and using that is not required to specify the boolean type

Comment (jm404) [reply tests/system/fim/common_tasks/verify_alerts_elasticsearch.py]: I read related documentation, you are completely right, thanks for the references!

**Pull request merged**